cmd/federation.go

```diff
@@ -11,17 +11,24 @@ import (
 )
 
 func main() {
+	// bootstrap logger
+	logger := log.NewLogger()
+
 	// Load connection string from .env file
 	err := godotenv.Load()
 	if err != nil {
-		log.Warn(fmt.Sprintf("failed to load env, %v", err))
+		logger.Warn(fmt.Sprintf("failed to load env, %v", err))
 	}
+
 	conn, err := db.Connect()
+	defer conn.Close()
```

Comment:
Hmm... I think the other line I commented on is better no? Shouldn't we check whether an error occurred before trying to defer closing? (not sure if the Fatal logger will still call deferred functions and cause more errors?)